engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -917,6 +931,11 @@ public void addInputPort(InputPort<?> portObject, Field field, InputPortFieldAnn
         metaPort.adqAnnotation = adqAnnotation;
         inPortMap.put(portObject, metaPort);
         markInputPortIfHidden(metaPort.getPortName(), metaPort, field.getDeclaringClass());
+        if (metaPort.getStreamCodec() == null) {
+          metaPort.setStreamCodec(portObject.getStreamCodec());
+        } else if (portObject.getStreamCodec() != null) {
+          LOG.info("Input port {} overwrites codec {} with {}.", metaPort, metaPort.getStreamCodec(), portObject.getStreamCodec());
```

Comment:
Yes, arguments should be flipped :). Apex does not log partitioner overwrite, though it may be good to have it logged. As logging is done once for a port, keeping log level to INFO.